### DISCUSSION TITLE
Access And Change fields value

### DISCUSSION THREAD

**Discussion by mikafumiDev** [vfsfitvnm/frida-il2cpp-bridge#254] [General]
![IMG_20230204_082007_643.jpg](https://user-images.githubusercontent.com/122904619/216734221-8f5f4870-01dc-4206-a78e-778680a201ed.jpg)

can we access or trace fields value and change it?

**Comment by berkedel** [comment]
To access the instance field
```
const cost = this.filed<Il2Cpp.Object>("_umpCost").value
```
From the snippet above the type of the value will be in `SecuredInt`, to update the value you need to invoke the update method from the `SecuredInt` class

**Comment by aimardcr** [comment]
Following solution by @berkedel, for `SecuredInt` there's a static function to create new `SecuredInt` object named `op_Implicit(System.Int32 value)` in the `SecuredInt` class.

For example: 
```
const SecuredInt = assembly.class('SecuredInt');
this.field('_umpCost').value = SecuredInt.method('op_Implicity').invoke(9999);
```

**Comment by datfooldive** [comment]
![Screenshot_20230211-102904817.jpg](https://user-images.githubusercontent.com/110718021/218255481-289acf06-3e14-49fe-a7e7-2494b7d6cf85.jpg)

so we need find method that fields store

what about this?

**Reply by aimardcr** [reply]
Can you elaborate your question further?

**Reply by datfooldive** [reply]
> Can you elaborate your question further?

so to change the value of a field we must determine from which method the field is stored
and make changes in that method

like
```
...class("foo").method("bar").implementation = (x) => {
    this.field("blabla").value = ....
    this.method("bar").invoke(x)
}
```

sorry I can't understand much so this is what I can catch

beginner and I only rely on google translate
